### ISSUE TITLE
Fiasp duration should be 180

### ISSUE THREAD

**Issue by aisaila** [NightscoutFoundation/xDrip#1259]
Hi all,

I think there is a problem with the current FIASP insulin duration time. 300 is to long and I think it should be somewhere near 180.

**Comment by vytux-com** [comment]
Where did you get this info from? I find that it's working way past when xdrip is predicting

**Comment by aisaila** [comment]
I saw this after the use with a pump. My doctor confirmed this. I am not saying this is the same for all users but maybe we can have a way to edit the duration setting and not have it hard coded into a .txt file.

**Comment by kjirvine** [comment]
> Hi all,
> 
> I think there is a problem with the current FIASP insulin duration time. 300 is to long and I think it should be somewhere near 180.

I have mine set at 150(2.5 hrs.)

**Comment by aisaila** [comment]
@kjirvine  How can you set it to a different value? In the config insulin profiles there are only check boxes, no configure option.

**Comment by kjirvine** [comment]
[Image]

[Image]

In profile editor upper right corner hamburger

Duration of insulin activity part of stored profile[Image]



________________________________

**Comment by ANDREAPIPPI** [comment]
Hi kjirvine, sorry, images are missing.
Pippo

**Comment by aisaila** [comment]
Sorry for this but as ANDREAPIPPI said, the images are missing and I could not find menu that you are talking about.

**Comment by kjirvine** [comment]
Sorry,

My setup is in nightscout and not xdrip
There is no selection of duration in xdrip just insulin type

Sorry
________________________________

**Comment by aisaila** [comment]
No problem. So this should be a good addition to xdrip, to have a way to configure insulin type params.

**Comment by 4RK4N** [comment]
you can set insulin duration: settings > xdrip+ predictive simulation settings > low level prediction values > insulin duration in hours

**Comment by kjirvine** [comment]
Thanks,
Forgot about that
Great help
________________________________

**Comment by aisaila** [comment]
@4RK4N yes I have that setup but it does not work with different insulin types. It only works with the default type. I wanted to have a way to modify the fiasp insulin type duration.

**Comment by Kustomwireless** [comment]
But how do you set those parameters if you are using and wanting to track more than 1 type of insulin? I have been trying to set up 2 types insulin tracking for my wife and do not see how to do that. I know where u can select the different types.

**Comment by kjirvine** [comment]
Make sure u have the most updated pre release version from March 11 2020

In predictive simulation
Multiple insulin type
Configure insulin type
Check the two different insulin type used
 Just below the check mark section
Select basal profile
Select bolus profile


________________________________

**Comment by tolot27** [comment]
This issue is related to #351.

**Comment by tolot27** [comment]
> But how do you set those parameters if you are using and wanting to track more than 1 type of insulin? I have been trying to set up 2 types insulin tracking for my wife and do not see how to do that. I know where u can select the different types.

That could be a new feature request.

@aisaila Is your problem solved with the low level prediction values setting?

**Comment by aisaila** [comment]
@tolot27 I've tried but it does not work, I have it set to 3h and it is displayed with 4.5h

**Comment by Navid200** [comment]
@aisaila @tolot27 
Would you please edit the title to reflect editing insulin type parameters?

**Comment by vguttmann** [comment]
Why don't we take this a bit further and allow users to add their own insulin?

Technically, since only three types can be used at once, editing the other ones should also work. It doesn't feel as nice, but it would probably also serve the purpose.

**Comment by Electrik-rich546456** [comment]
I have the same problem with lantus it predicts is off by many hours for each dose. It lasts 23ish hours not the 5 hours it shows on the graph.
